tortoise/models.py

```diff
@@ -1054,17 +1054,17 @@ async def get_or_create(
         if not defaults:
             defaults = {}
         db = using_db or cls._choose_db(True)
-        async with in_transaction(connection_name=db.connection_name) as connection:
+        try:
+            return await cls.filter(**kwargs).using_db(db).get(), False
+        except DoesNotExist:
             try:
-                return (
-                    await cls.select_for_update().filter(**kwargs).using_db(connection).get(),
-                    False,
-                )
-            except DoesNotExist:
-                try:
+                async with in_transaction(connection_name=db.connection_name) as connection:
```

Comment:
Do we need transaction here for just single create?

Reply:
Probably not. But this logic is in django.

https://github.com/django/django/blob/main/django/db/models/query.py#L938

Reply:
Seems like this sub-transaction is used, so that in case of integrity error here - it aborts it's transaction, so we should have one here for it to be aborted, otherwise it can abort whole parent transaction that is out of scope of this function, which would lead for unexpected behaviour for user

Let's make similar transaction here, as it was in your previous version

Reply:
Done